src/borg/patterns.py

```diff
@@ -150,9 +150,9 @@ def match(self, path):
 
         if value is not non_existent:
             # we have a full path match!
-            # TODO: get from pattern; don't hard-code
-            self.recurse_dir = True
-            return value
+            self.recurse_dir = (value is IECommand.ExcludeNoRecurse)
```

Comment:
This looks like it's inverted, setting recurse_dir to True when value says Exclude*NoRecurse*.